onnx/defs/math/defs.cc

```diff
@@ -2006,6 +2007,301 @@ ONNX_OPERATOR_SET_SCHEMA(
+            AttributeProto::INT,
+            OPTIONAL_VALUE)
+        .Attr(
+            "N",
+            "Size of each output feature. If not present, should be inferred from input B.",
+            AttributeProto::INT,
+            OPTIONAL_VALUE)
+        .Attr(
+            "accuracy_level",
+            "The minimum accuracy level of input A, can be: "
+            "0(unset), 1(float), 2(float16), 3(Bfloat16), 4(int8) (default unset). It is used to control how input A "
+            "is quantized or downcast internally while doing computation, for example: 0 means input A will not be "
+            "quantized or downcast while doing computation, 4 means input A can be quantized with the same block_size"
+            "to int8 internally from type T1.",
+            AttributeProto::INT,
+            static_cast<int64_t>(0))
+        .Attr(
+            "bits",
+            "Number of bits used for weight quantization (default 4)",
+            AttributeProto::INT,
+            static_cast<int64_t>(4))
+        .Attr(
+            "block_size",
+            "Number of groupsize used for weight quantization (default 128)."
+            "It needs to be a power of 2 and not smaller than 16",
+            AttributeProto::INT,
+            static_cast<int64_t>(128))
+        .Input(
+            0,
+            "A",
+            "A 2 dimensional input tensor, not quantized, with shape [M][K]",
+            "T1",
+            OpSchema::Single,
+            true,
+            1,
+            OpSchema::NonDifferentiable)
+        .Input(
+            1,
+            "B",
+            "A 3 dimensional input tensor of shape [N][n_blocks_per_col][blob_size], "
+            "i.e. an N by 2 dimensional data blob containing the packed B bits",
+            "T2",
+            OpSchema::Single,
+            true,
+            1,
+            OpSchema::NonDifferentiable)
+        .Input(
+            2,
+            "scales",
+            "Tensor of quantization scales. It should have shape [N * n_blocks_per_col]",
+            "T1",
+            OpSchema::Single,
+            true,
+            1,
+            OpSchema::NonDifferentiable)
+        .Input(
+            3,
+            "zero_points",
+            "Tensor of quantization zero points. "
+            "If T3 is the same as T1 it should have shape [N * n_blocks_per_col]. "
+            "If T3 is the same as T2 is should have shape [N * CeilDiv(n_blocks_per_col * bits, 8)]",
+            "T3",
+            OpSchema::Optional,
+            true,
+            1,
+            OpSchema::NonDifferentiable)
+        .Input(
+            4,
+            "bias",
+            "Bias to add to result. It should have shape [N]",
+            "T1",
+            OpSchema::Optional,
+            true,
+            1,
+            OpSchema::NonDifferentiable)
+        .Output(
+            0,
+            "Y",
+            "Matrix multiply results. The output tensor has the same rank as the input.",
```

Comment:
Maybe say this is of shape [M][N] ? As I understand it, this supports only 2D inputs, without support for broadcast. So, no need for the second line about "same rank as the input".